src/Controls/src/Core/Platform/Windows/Extensions/CollectionViewExtensions.cs

```diff
@@ -13,13 +13,20 @@ public static class CollectionViewExtensions
 			if (layout is null)
 				return null;
 
-			var h = layout?.ItemSpacing ?? 0;
-			var v = layout?.ItemSpacing ?? 0;
-			var margin = WinUIHelpers.CreateThickness(h, v, h, v);
-
 			var style = new WStyle(typeof(GridViewItem));
 
-			style.Setters.Add(new WSetter(FrameworkElement.MarginProperty, margin));
+			if (layout?.Orientation == ItemsLayoutOrientation.Vertical)
+			{
+				style.Setters.Add(new WSetter(FrameworkElement.MinHeightProperty, 0));
+
+				var h = layout?.ItemSpacing ?? 0;
+				var v = layout?.ItemSpacing ?? 0;
+				var margin = WinUIHelpers.CreateThickness(h, v, h, v);
+				style.Setters.Add(new WSetter(FrameworkElement.MarginProperty, margin));
```

Comment:
Why do we only set Margin on Vertical ? what happens if i give ItemsSpacing in a Horizontal layout ?

Reply:
Need to do more tests to know exactly why, but could have sense to use in both cases. The code comes from XF https://github.com/xamarin/Xamarin.Forms/blob/4f3acdfb0bc98e3ba8e2f3eae86e88602350ee84/Xamarin.Forms.Platform.UAP/CollectionView/StructuredItemsViewRenderer.cs#L241

Reply:
It looks like the test is broken because in [this PR](https://github.com/dotnet/maui/pull/14532/files#diff-c1fe5c4ab4be56e26f1b4f639462b6e27e7e807f21a73dddc2e19213dc514a5b) when the methods were moved from the handler to the extension method class, the MinHeight and MinWidth property settings weren't moved along with everything else. The code that was removed just needs to be added back in, the rest of this is unnecessary.